enhancements/user-defined-default-ingress-controller.md

```diff
@@ -0,0 +1,273 @@
+---
+title: user-defined-default-ingress-controller
+authors:
+  - "@ironcladlou"
+reviewers:
+  - "@ironcladlou"
+  - "@Miciah"
+  - "@danehans"
+approvers:
+  - "@ironcladlou"
+  - "@Miciah"
+  - "@danehans"
+creation-date: 2019-09-08
+last-updated: 2019-09-09
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# User-defined Default Ingress Controller
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+This proposal is to enable users to define the default
+IngressController<sup>[1](https://docs.openshift.com/container-platform/4.1/networking/configuring-ingress-cluster-traffic/configuring-ingress-cluster-traffic-ingress-controller.html#configuring-ingress-cluster-traffic-ingress-controller),[2](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go)</sup>
+during OpenShift installation.
+
+Changes to the OpenShift [Installer](https://github.com/openshift/installer) and
+[Ingress Operator](https://github.com/openshift/cluster-ingress-operator) are
+proposed which will provide the basis for a supportable customization procedure.
+The procedure is proposed for incorporation into the OpenShift
+[documentation](https://github.com/openshift/openshift-docs).
+
+## Motivation
+
+OpenShift's default IngressController is automatically created based on the
+cluster platform and cluster-global configuration. The default is reliable for
+installation, but typically requires reconfiguration soon after installation.
+
+However, there are some updates to an existing IngressController which are
+disruptive or destructive and which users would prefer to configure during
+installation.
+
+Because of the broad range of ingress needs, it's difficult to anticipate what
+subset of IngressController API fields should be exposed for configuration at
+the time of installation  Allowing the user to replace the default
+IngressController in its entirety provides maximum expressiveness and
+flexibility without introducing new configuration API surface area.
+
+### Goals
+
+* Create a procedure which safely allows users to provide a specific default
+  IngressController resource to the installer.
+
+### Non-Goals
+
+* No changes to the `config.openshift.io` or `operator.openshift.io` APIs are
+  proposed.
+
+## Proposal
+
+The following specific changes are proposed.
+
+1. Add a `render` command to the Ingress Operator binary which emits the
```

Comment:
Based on speaking with @abhinavdahiya, rendering happens on the bootstrap machine.  You'll need to go a step back to here: https://github.com/openshift/installer/blob/master/pkg/asset/manifests/ingress.go .  

For the short term, you could copy the CRD yaml and after https://github.com/openshift/enhancements/pull/41 lands it can be vendored.